inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/RedisSource.java

```diff
@@ -74,7 +162,18 @@ protected boolean isRunnable() {
 
     @Override
     protected void releaseSource() {
-
+        LOGGER.info("releasing redis source");
+        synchronized (this) {
```

Comment:
There won't be multi-threaded calls here, no need to lock

Reply:
done